samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,192 @@
+endif()
+
+if ((NOT NANOPB_BIN) OR (NOT PROTOC_BIN))
+    message(STATUS "nanopb not found.")
+    message(STATUS "Please download nanopb from here https://jpa.kapsi.fi/nanopb/download/")
+    message(STATUS "and extract to either this sample directory or somewhere cmake can find it via system PATH.")
+    message(FATAL_ERROR "nanopb required by this project.")
+endif()
+
+message(STATUS "Using protoc: ${PROTOC_BIN}")
+message(STATUS "Using protoc-gen-nanopb: ${NANOPB_BIN}")
+
+if(ALEXA_EMBEDDED_PATH)
+  # Do nothing, just use the provided path
+elseif(WEST)
+  ## Use `west list` to find the alexa-embedded path
+  execute_process(
+    COMMAND
+    ${WEST} list Alexa-Gadgets-Embedded-Sample-Code --format={posixpath}
+    OUTPUT_VARIABLE
+    west_project_output
+    RESULT_VARIABLE
+    result
+  )
+  string(REGEX REPLACE "[\r\n]+" "" ALEXA_EMBEDDED_PATH "${west_project_output}")
+  if(${result})
+    message(FATAL_ERROR "Failed to find alexa-embedded, cannot build sample")
+  endif()
+else()
+  message(FATAL_ERROR "west not installed, please provide ALEXA_EMBEDDED_PATH to CMake to build this sample")
+endif()
+
+if(NANOPB_PATH)
+  # Do nothing, just use the provided path
+elseif(WEST)
+  ## Use `west list` to find the nanopb path
+  execute_process(
+    COMMAND
+    ${WEST} list nanopb --format={posixpath}
+    OUTPUT_VARIABLE
+    west_project_output
+    RESULT_VARIABLE
+    result
+  )
+  string(REGEX REPLACE "[\r\n]+" "" NANOPB_PATH "${west_project_output}")
+  if(${result})
+    message(FATAL_ERROR "Failed to find nanopb source files, cannot build sample")
+  endif()
+else()
+  message(FATAL_ERROR "west not installed, please provide NANOPB_PATH to CMake to build this sample")
+endif()
+
+function(compile_protobufs_in_dir dir recursive)
+    # Find all .proto files and include directories
+    if (recursive)
+        file(GLOB_RECURSE PROTO_FILES "${dir}/*.proto")
+    else()
+        file(GLOB PROTO_FILES "${dir}/*.proto")
+    endif()
+
+    foreach(file ${PROTO_FILES})
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
+    endforeach()
+
+    # Compile .proto files in their respective directories
+    foreach(file ${PROTO_FILES})
+        # Add all directories containing .proto files.
+        # Duplicates and current directory is then removed from "-I" arguments
+        set(PROTOC_ARGS "${PROTOC_INC_DIRS}")
+        get_filename_component(FILE_PATH ${file} DIRECTORY)
+        get_filename_component(FILE_NAME ${file} NAME)
+        list(REMOVE_ITEM PROTOC_ARGS "-I${FILE_PATH}")
+        list(REMOVE_DUPLICATES PROTOC_ARGS)
+        list(APPEND PROTOC_ARGS "-I.")
+        list(APPEND PROTOC_ARGS "--nanopb_out=${FILE_PATH}")
+        list(APPEND PROTOC_ARGS "${FILE_NAME}")
+        message(STATUS "Compiling proto definition: ${FILE_NAME}")
+        execute_process(
+            COMMAND ${PROTOC_BIN} ${PROTOC_ARGS}
+            WORKING_DIRECTORY ${FILE_PATH}
+        )
```

Comment:
seems the guide need some more information on how to get this to work.
I'm getting following error:
```
Failed to build nanopb_pb2.py: protoc --python_out=/projects/github/ncs/nanopb/generator/proto /projects/github/ncs/nanopb/generator/proto/nanopb.proto -I=/projects/github/ncs/nanopb/generator/proto -I/projects/github/ncs/nanopb/generator/proto
Failed to build nanopb_pb2.py: protoc --python_out=/projects/github/ncs/nanopb/generator/proto /projects/github/ncs/nanopb/generator/proto/nanopb.proto -I=/projects/github/ncs/nanopb/generator/proto -I/projects/github/ncs/nanopb/generator/proto
Traceback (most recent call last):
  File "/projects/github/ncs/nanopb/generator/nanopb_generator.py", line 50, in <module>
    from .proto import nanopb_pb2
ImportError: attempted relative import with no known parent package

```

Reply:
Are you using windows?

Reply:
nope, Linux.
and I guess if I were on Windows, this `--python_out=/projects/..`would probably start with `--python_out=/C/...` :wink:

Reply:
Ah yes, fair enough!
What is your setup with regards to protoc and nanopb? Did you download the compressed archive from the nanopb downloads page?